src/ringmesh/io/geomodel_builder_resqml.cpp

```diff
@@ -0,0 +1,968 @@
+
+                bool match = true;
+                for( auto b : range( region.nb_boundaries() ) )
+                {
+                    const Surface3D& surface = region.boundary( b );
+                    for( auto p : range( surface.nb_mesh_elements() ) )
+                    {
+                        auto center = surface.mesh_element_barycenter( p );
+                        auto result = nn_search.get_neighbors(
+                            center, surface.geomodel().epsilon() );
+                        if( result.empty() )
+                        {
+                            match = false;
+                            break;
+                        }
+                    }
+                    if( !match )
+                    {
+                        break;
+                    }
+                }
+                if( match )
+                {
+                    region_index = (int) r;
+                    break;
+                }
+            }
+
+            if( region_index == -1 )
+            {
+                if( !geomodel_.entity_type_manager()
+                         .mesh_entity_manager.is_valid_type(
+                             Region3D::type_name_static() )
+                    || unstructuredGridRepSet.size() > 1 )
+                {
+                    return false;
+                }
+                else
+                {
+                    region_index = 0;
+                }
+            }
+
+            // corresponding region found, build its volume mesh
+            const gmme_id region_id(
+                region_type_name_static(), (index_t) region_index );
+
+            auto mesh_builder =
+                builder_.geometry.create_region_builder( region_id.index() );
+
+            for( auto v : range( mesh->nb_vertices() ) )
+            {
+                mesh_builder->create_vertex( mesh->vertex( v ) );
+            }
+
+            for( auto cell : range( mesh->nb_cells() ) )
+            {
+                mesh_builder->create_cells(
+                    (index_t) 1, mesh->cell_type( cell ) );
+
+                const index_t nb_vertices = mesh->nb_cell_vertices( cell );
+                std::vector< index_t > cell_vertices( nb_vertices, 0 );
+                for( auto v : range( nb_vertices ) )
+                {
+                    ElementLocalVertex lv( cell, v );
+                    mesh_builder->set_cell_vertex(
+                        lv, mesh->cell_vertex( lv ) );
+                }
+            }
+
+            mesh_builder->connect_cells();
+
+            // property
+            const Region3D& cur_reg = geomodel_.region( region_id.index() );
+
+            for( auto prop_index :
+                range( unstructured_grid->getValuesPropertySet().size() ) )
+            {
+                AbstractValuesProperty* propVal =
+                    unstructured_grid->getValuesPropertySet()[prop_index];
+                // TODO there is only one patch in a UnstructuredGrid
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1135)